pkg/gitops/init.go

```diff
@@ -123,52 +121,48 @@ func SetupRepo(k8s *K8sClient, mode StorageModeType, dir string) (*CustomRepo, e
 	if err := cr.AddAndCommit("audit-init", "system@audit.antrea.io", "Initial commit of existing policies"); err != nil {
 		return nil, fmt.Errorf("unable to add/commit existing reosurces to repository: %w", err)
 	}
-	klog.V(2).Infof("repository successfully initialized at %s", cr.Dir)
+	klog.V(2).Infof("repository successfully initialized at %s", dir)
 	return &cr, nil
 }
 
-func (cr *CustomRepo) createRepo(storer *memory.Storage) (*git.Repository, error) {
-	if cr.StorageMode == StorageModeInMemory {
-		r, err := git.Init(storer, cr.Fs)
-		if err == git.ErrRepositoryAlreadyExists {
-			klog.V(2).InfoS("resource repository already exists - skipping initialization")
-			return nil, err
-		} else if err != nil {
-			return nil, fmt.Errorf("unable to initialize git repo %w: ", err)
-		}
-		return r, nil
-	}
-	if cr.Dir == "" {
-		path, err := os.Getwd()
-		if err != nil {
-			return nil, fmt.Errorf("unable to retrieve the current working directory: %w", err)
-		}
-		if path != "/" {
-			cr.Dir = path
+func setupStorage(dir string, mode StorageModeType) (storage.Storer, billy.Filesystem, error) {
+	var storer storage.Storer
+	var worktreeFs, storerFs billy.Filesystem
+	if mode == StorageModeDisk {
+		if dir == "" {
+			dir, _ = os.Getwd()
 		}
+		dir += "/resource-auditing-repo/"
+		worktreeFs = osfs.New(dir)
+		storerFs = osfs.New(dir + "/.git")
```

Comment:
can you use the standard `Join` function to build paths: https://pkg.go.dev/path#Join